packages/perseus-editor/src/components/__tests__/interactive-graph-settings.test.tsx

```diff
@@ -227,7 +231,12 @@ describe("InteractiveGraphSettings", () => {
         render(<InteractiveGraphSettings onChange={onChange} />);
 
         // Act
-        const input = screen.getByRole("textbox", {name: "x Range"});
+        // Note: The textbox's `name` attribute is "x Range 10" because it's
+        // encased in a <label> element, which applies everything surrounding
+        // the first element to the first element's label. This means it
+        // includes the "10" from the second textbox. Same for the other
+        // RangeInput tests below.
+        const input = screen.getByRole("textbox", {name: "x Range 10"});
```

Comment:
After some more testing weirdness, I really do think we should switch to unique IDs at some point. I might do it in a following PR? Or maybe I'll wait for a larger refactor in case we can get to a point with functional components so I can use the hook for it.

Reply:
This has to do with the fact that RangeInput is actually two text fields wrapped under one label. Only the first text field receives the label, and it treats everything around it as a label (both the actual text and the second text field).

The way to get around this is to give the desired text field an ID and set the label's `htmlFor` to that ID, but this is a problem if there are two interactive graph editors because you can't have multiple IDs. The second label will point to the first instance of the text field. So what we'd need to do is use a unique ID provider. Which I _can_ do here with the unique ID HOC. Honestly I might just do that...

Reply:
I just really don't like using HOCs when hooks are so much better.